data_processing/wdl/scatterGuppy.wdl

```diff
@@ -0,0 +1,348 @@
+	scatter (fast5_tar in fast5_tar_files) {
+		call splitFast5s {
+			input:
+				file_to_split_tar = fast5_tar,
+				desired_size_GB = desired_size_GB
+
+		}
+
+		# call guppyGPU on each of the smaller "split" tar files
+		scatter (split_fast5_tar in splitFast5s.split_fast5_tar) {
+			call guppyGPU {
+				input:
+					fast5_tar_file = split_fast5_tar
+			}
+		}
+
+		call concatenateBam {
+			input:
+				files = guppyGPU.pass_bam,
+				sample_name = sample_name,
+				guppy_version = guppy_version
+		}
+
+		call concatenateFastq {
+			input:
+				files = guppyGPU.pass_fastq,
+				sample_name = sample_name,
+				guppy_version = guppy_version
+		}
+
+		call concatenateSummary {
+			input:
+				files = guppyGPU.summary,
+				sample_name = sample_name,
+				guppy_version = guppy_version
+		}
+
+	}
+
+	output {
+		Array[File] bams = concatenateBam.concatenatedBam
+		Array[File] fastqs = concatenateFastq.concatenatedFastq
+		Array[File] summaries = concatenateSummary.concatenatedSummary
+	}
+	
+}
+
+task concatenateBam {
+	input {
+		Array[File] files
+		
+		String sample_name
+		String guppy_version
+
+		String dockerImage = "tpesout/megalodon:latest"
+
+		Int preempts = 3
+		Int memSizeGB = 8
+		Int threadCount = 3
+		Int diskSizeGB = 500
+	}
+	
+
+	command {
+		samtools merge -o "${sample_name}_${guppy_version}.bam" ${sep=" " files}
+	}
+
+	output {
+		File concatenatedBam = "${sample_name}_${guppy_version}.bam"
+	}
+
+	runtime {
+		memory: memSizeGB + " GB"
+		cpu: threadCount
+		disks: "local-disk " + diskSizeGB + " SSD"
+		docker: dockerImage
+		preemptible : preempts
+	}
+}
+
+
+task concatenateFastq {
+	input {
+		Array[File] files
+		String sample_name
+		String guppy_version
+
+
+		String dockerImage = "tpesout/megalodon:latest"
+
+		# runtime
+		Int preempts = 3
+		Int memSizeGB = 8
+		Int threadCount = 3
+		Int diskSizeGB = 500
+	}
+	
+
+	command {
+		cat ${sep=" " files} | gzip -c > "${sample_name}_${guppy_version}.fastq.gz"
+	}
+
+	output {
+		File concatenatedFastq = "${sample_name}_${guppy_version}.fastq.gz"
+	}
+
+	runtime {
+		memory: memSizeGB + " GB"
+		cpu: threadCount
+		disks: "local-disk " + diskSizeGB + " SSD"
+		docker: dockerImage
+		preemptible : preempts
+	}
+}
+
+task concatenateSummary {
+	input {
+		Array[File] files
+		String sample_name
+		String guppy_version
+
+
+		String dockerImage = "tpesout/megalodon:latest"
+
+		# runtime
+		Int preempts = 3
+		Int memSizeGB = 8
+		Int threadCount = 3
+		Int diskSizeGB = 500
+	}
+	
+
+	command {
+		cat ${sep=" " files} > "tmp.txt"
+		# remove duplicate headers
+		awk 'NR==1 || !/^filename/' "tmp.txt" > "${sample_name}_${guppy_version}_sequencing_summary.txt"
+	}
+
+	output {
+		File concatenatedSummary = "${sample_name}_${guppy_version}_sequencing_summary.txt"
+	}
+
+	runtime {
+		memory: memSizeGB + " GB"
+		cpu: threadCount
+		disks: "local-disk " + diskSizeGB + " SSD"
+		docker: dockerImage
+		preemptible : preempts
+	}
+}
+
+
+task splitFast5s {
```

Comment:
Not a big deal, I wouldn't change anything right now, but it would be great for the task definitions to be in the same order they are called.